### ISSUE TITLE
use doxygen to generate API documentation as part of the automated build

### ISSUE THREAD

**Issue by cchriste** [SCIInstitute/ShapeWorks#464]
We are incorporating doxygen-style comments* within our API classes (Mesh, Image, etc), and this can be a portion of our developer documentation for using the ShapeWorks API.

```
/// functionName
///
/// description of the function
///
/// \param param1 description of param1
/// ...
bool functionName(int param1, ...) ...
```

Be sure to add these to the .cpp not the .h files, enabling a very clean header and at-a-glance understanding of what functions exist in a class. Plus they're kind of ugly and verbose, but they're what doxygen requires.

**Comment by cchriste** [comment]
Closely related to #1079.

**Comment by cchriste** [comment]
Resolved by PR #1704 (issue #1575)
